aten/src/ATen/native/TensorShape.cpp

```diff
@@ -545,6 +545,23 @@ std::vector<Tensor> tensor_split(const Tensor& self, IntArrayRef indices, int64_
   return splits;
 }
 
+std::vector<Tensor> tensor_split(const Tensor& self, const Tensor& tensor_indices_or_sections, int64_t dim) {
+  TORCH_CHECK(self.dim() > 0, "expected at least a 1-dimensional tensor");
+  TORCH_CHECK(tensor_indices_or_sections.device() == kCPU,
+    "expected tensor_indices_or_sections to be on cpu")
+  if (tensor_indices_or_sections.dim() == 0) {
+    int64_t sections = tensor_indices_or_sections.item<int64_t>();
+    return self.tensor_split(sections, dim);
+  } else {
+    TORCH_CHECK(tensor_indices_or_sections.dim() == 1 &&
+      tensor_indices_or_sections.scalar_type() == at::kLong,
```

Comment:
This check for kLong is good but should happen immediately before or after the check that `tensor_indices_or_sections` is on the CPU. Otherwise a scalar float tensor could be passed.

Reply:
The check for dimensions can also happen along with the other checks. Check that the dimension is <= 1, and if it's not say it's expected to be zero or one dimensions and print how many dimensions it actually is, just like the comment for the device warning above.